pkg/schema/validator.go

```diff
@@ -35,6 +35,105 @@ const (
 	reservedPropConnections = "connections"
 )
 
+// joinPath concatenates two path segments with a dot separator for property path tracking.
+func joinPath(parent, child string) string {
+	if parent == "" {
+		return child
+	}
+	if child == "" {
+		return parent
+	}
+	return parent + "." + child
+}
+
+// isPlatformOptionsPath checks if a path ends with "platformOptions" (last segment after splitting by dots).
+func isPlatformOptionsPath(path string) bool {
+	if path == "" {
+		return false
+	}
+	segments := strings.Split(path, ".")
+	return segments[len(segments)-1] == "platformOptions"
+}
+
+// isPlatformOptionsAdditionalPropertiesPath checks if a path points to the
+// additionalProperties schema under a platformOptions property.
+func isPlatformOptionsAdditionalPropertiesPath(path string) bool {
+	if path == "" {
+		return false
+	}
+	segments := strings.Split(path, ".")
+	if len(segments) < 2 {
+		return false
+	}
+	return segments[len(segments)-2] == "platformOptions" && segments[len(segments)-1] == "additionalProperties"
+}
+
+// normalizePlatformOptionsAny rewrites type: any occurrences that sit beneath
+// platformOptions.additionalProperties into spec-compliant schemas so the
+// OpenAPI validator accepts them while keeping the rest of the document intact.
+func normalizePlatformOptionsAny(schema *openapi3.Schema) {
+	normalizePlatformOptionsAnyWithPath(schema, "")
+}
+
+func normalizePlatformOptionsAnyWithPath(schema *openapi3.Schema, path string) {
```

Comment:
OpenAPI doesnt support the any type so we remove it and perform our own validation